### ISSUE TITLE
Remove (immediately and with all haste) `IResolver.lookupRecord`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6091]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6091|
|Type|enhancement|
|Created|2012-10-10 13:04:49Z|
|Branch|https://github.com/twisted/twisted/tree/remove-lookuprecord-6091|

This method of the `IResolver` interface has no implementation in Twisted and has never had an implementation in Twisted.  It is a documentation error.

I propose that this is acceptable without deprecation because:

  * Twisted has never included an implementation of this method for applications to call.
  * Twisted has never called this method on any object, either defined by Twisted or supplied by an application.
  * It is redundant with `query`.
  * Any application code which was implementing this method for some reason of its own will be unaffected by the removal of the method from the interface.

<details><summary>Searchable metadata</summary>

```
trac-id__6091 6091
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_remove_lookuprecord_6091 branches/remove-lookuprecord-6091
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349874289000000 1349874289000000
changetime__1349896463000000 1349896463000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [36047]) Branching to 'remove-lookuprecord-6091'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @itamarst|
|-|-|

There, I removed it.  No deprecation or anything, just **bam** gone.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

I like it. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [36069]) Merge remove-lookuprecord-6091

Author: exarkun
Reviewer: therve
Fixes: [#6091](https://github.com/twisted/twisted/issues/6091)

Remove `twisted.internet.interfaces.IResolver.lookupRecord`, a bogus method that was never
implemented or used.
